tests/functional/test_upgrade_actions.py

```diff
@@ -215,3 +217,45 @@ def _extract_configs(cluster: Cluster):
             for service in cluster.service_list()
         },
     }
+
+
+class TestUpgradeActionRelations:
+    """Test cases when upgrade action"""
+
+    @pytest.mark.parametrize(
+        ("folder_dir", "file_dir"),
+        [
+            ('upgrade_failures', 'before_switch'),
+            ('upgrade_success', 'after_switch'),
+        ],
+    )
+    def test_check_upgrade_actions_relations(self, sdk_client_fs, folder_dir, file_dir):
+        """
+        Test bundle action fails before bundle_switch was performed
+        """
+        bundle = sdk_client_fs.upload_from_fs(get_data_dir(__file__, folder_dir, 'old'))
+        cluster = bundle.cluster_create('Test Cluster for Upgrade')
+        sdk_client_fs.upload_from_fs(get_data_dir(__file__, folder_dir, file_dir))
+        with allure.step("Check cluster actions list before update"):
+            actions_before = cluster.action_list().data
+            assert len(actions_before) == 1, "Should be 1 action"
+            assert actions_before[0].display_name == "dummy_action", "Should be action 'dummy_action'"
+        cluster.upgrade().do().wait()
+        cluster.reread()
+        with allure.step("Check jobs"):
+            jobs = sdk_client_fs.job_list()
+            assert len(jobs) >= 3, "There are should be 3 or more jobs"
+            assert len(
+                set(['first action after switch', 'switch action', 'first_action'])
+                & set([j.display_name for j in jobs.data])
+            ), "Jobs names differ"
+            assert (
+                len([log for data in jobs.data for log in data.log_files]) >= 6
+            ), "There are should be 6 or more log files"
+        with allure.step("Check cluster actions list after update"):
+            actions_after = cluster.action_list().data
+            assert len(actions_after) == 2, "Should be 2 actions"
+            assert [action.display_name for action in actions_after] == [
+                'dummy_action',
+                'restore',
+            ], "Should be actions 'dummy_action' and 'restore'"
```

Comment:
Maybe use sets here, because we don't want to rely on order of actions in list provided by ADCM?

Reply:
changed